src/features/persoon/persoon-zoeken/PersonenOverzicht.vue

```diff
@@ -49,29 +49,47 @@
 <script lang="ts" setup>
 import DutchDate from "@/components/DutchDate.vue";
 import type { Persoon } from "@/services/brp";
-import type { Klant } from "@/services/klanten";
 import { useRouter } from "vue-router";
-import { ensureKlantForBsn } from "./ensure-klant-for-bsn";
+import { useOrganisatieIds } from "@/stores/user";
 import { mutate } from "swrv";
 import { watchEffect } from "vue";
 
+import { ensureKlantForBsn as ensureKlantForBsn1 } from "@/services/openklant1/service";
+import { ensureKlantForBsn as ensureKlantForBsn2, useOpenKlant2 } from "@/services/openklant2/service";
+import type { Klant as Klant1 } from "@/services/openklant1/types";
+import type { Klant as Klant2 } from "@/services/openklant2/types";
+
 const props = defineProps<{
   records: Persoon[];
   navigateOnSingleResult?: boolean;
 }>();
 
 const router = useRouter();
 
+const isOpenKlant2 = await useOpenKlant2();
+
+type Klant = Klant1 | Klant2;
+
 const getKlantUrl = (klant: Klant) => `/personen/${klant.id}`;
 
 const navigate = async (persoon: Persoon) => {
   const { bsn } = persoon;
-  if (!bsn) throw new Error();
-  const klant = await ensureKlantForBsn({
-    bsn,
-  });
+  if (!bsn) throw new Error("BSN is required");
+
+  let klant: Klant;
+
+  if (isOpenKlant2) {
```

Comment:
ik zou de structuur net even anders doen. 
er zit nu businesslogica in de service (find or create klant) en het vue component (gebruik openklant 1 of 2). 
De keuze welke openklant versie gebruikt moet worden en of de klant nog aangemaakt moet worden zou ik in een service.ts onder persoon onderbrengen.